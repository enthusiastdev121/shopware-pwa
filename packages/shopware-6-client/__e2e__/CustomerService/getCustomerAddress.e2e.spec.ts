import {
  update,
  login,
  getCustomerAddress,
  createCustomerAddress,
  deleteCustomerAddress,
} from "@shopware-pwa/shopware-6-client";
import { address, name } from "faker";

describe("shopware-6-client - E2E - CustomerService - getCustomerAddress", () => {
  beforeEach(() => {
    update({ contextToken: "" });
  });

  it("should fetch customer address for given address id", async () => {
    await login({ username: "test.e2e@test.pl", password: "password" });
    const customerAddress: any = await createCustomerAddress({
      firstName: name.firstName(),
      lastName: name.lastName(),
      salutationId: "c370eb5cd1df4d4dbcc78f055b693e79",
      zipcode: address.zipCode(),
      city: address.city(),
      street: address.streetName(),
      salutation: "Mr",
      countryId: "38245a84c3d5425b8bac97fc845b5ddd",
    });
    const result: any = await getCustomerAddress(customerAddress.data);
    await deleteCustomerAddress(customerAddress.data);
    result._uniqueIdentifier = "mockedUniqueIdentifier";
    result.city = "mockedCity";
    result.createdAt = "mockedCreatedAt";
    result.zipcode = "mockedZipcode";
    result.street = "mockedStreet";
    result.firstName = "mockedFirstName";
    result.lastName = "mockedLastName";
    result.id = "mockedId";
    expect(result).toMatchSnapshot();
  });

  it("should return error for address id that not matches any address", async () => {
    try {
      await login({ username: "test.e2e@test.pl", password: "password" });
      await getCustomerAddress("qwa");
      expect("didn't throw an error").toEqual("should throw an error");
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });

  it("should not fetch address for unlogged user", async () => {
    try {
      await getCustomerAddress("ed9e8daeae9a46cf84abf4141fc5d106");
      expect("didn't throw an error").toEqual("should throw an error");
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });
});
