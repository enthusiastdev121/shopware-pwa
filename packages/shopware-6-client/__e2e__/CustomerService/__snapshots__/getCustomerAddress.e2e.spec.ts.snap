// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopware-6-client - E2E - CustomerService - getCustomerAddress should fetch customer address for given address id 1`] = `
Object {
  "_uniqueIdentifier": "mockedUniqueIdentifier",
  "additionalAddressLine1": null,
  "additionalAddressLine2": null,
  "city": "mockedCity",
  "company": null,
  "country": null,
  "countryId": "38245a84c3d5425b8bac97fc845b5ddd",
  "countryState": null,
  "countryStateId": null,
  "createdAt": "mockedCreatedAt",
  "customFields": null,
  "customer": null,
  "customerId": "d5ba879ae22f4dcd8f881046dbcedc41",
  "department": null,
  "extensions": Object {
    "foreignKeys": Object {
      "_uniqueIdentifier": null,
      "createdAt": null,
      "extensions": Array [],
      "translated": Array [],
      "updatedAt": null,
      "versionId": null,
    },
  },
  "firstName": "mockedFirstName",
  "id": "mockedId",
  "lastName": "mockedLastName",
  "phoneNumber": null,
  "salutation": null,
  "salutationId": "c370eb5cd1df4d4dbcc78f055b693e79",
  "street": "mockedStreet",
  "title": null,
  "translated": Array [],
  "updatedAt": null,
  "vatId": null,
  "versionId": null,
  "zipcode": "mockedZipcode",
}
`;

exports[`shopware-6-client - E2E - CustomerService - getCustomerAddress should not fetch address for unlogged user 1`] = `[Error: Request failed with status code 403]`;

exports[`shopware-6-client - E2E - CustomerService - getCustomerAddress should return error for address id that not matches any address 1`] = `[Error: Request failed with status code 400]`;
